### PR TITLE
DOCS-#5: Add a contributing page for unidist

### LINKED ISSUE #5: Add a contributing page for `unidist`
Should be similar to the contributing page in Modin - https://modin.readthedocs.io/en/latest/contributing.html.

### PR DESCRIPTION
**Pull request by YarShev**
Signed-off-by: Igoshev, Yaroslav <yaroslav.igoshev@intel.com>

## What do these changes do?

- [x] passes `flake8 .`
- [x] passes `black .`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5  
- [x] tests passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: docs/developer/contributing.rst [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+structure of the tests for ease of maintenance.
+
+Running the tests
+-----------------
+
+To run the entire test suite, run the following from the project root:
+
+.. code-block:: bash
+
+  pytest unidist/test
```
Comment (prutskov) [inline docs/developer/contributing.rst:152]:
```suggestion
  python unidist/cli -m pytest unidist/test
```
or
```suggestion
  unidist -m pytest unidist/test
```

We should describe a general way of running any unidist program to make it possible to run tests on any backend.

File: docs/developer/contributing.rst [hunk 2]
```diff
@@ -0,0 +1,185 @@
+..
+      Copyright (C) 2021 Modin authors
```
Comment (prutskov) [inline docs/developer/contributing.rst:2]:
```suggestion
      Copyright (C) 2021-2022 Modin authors
```

Comment (YarShev) [reply docs/developer/contributing.rst:2]: fixed

File: docs/developer/contributing.rst (hunk 1 above)
Comment (YarShev) [reply docs/developer/contributing.rst:152]: Fixed. Added a line about CLI and a comment on inserting a link to the CLI.

File: docs/developer/contributing.rst [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+.. code-block:: bash
+
+  pytest unidist/test
+
+If you've only modified a small amount of code, it may be sufficient to run a single test or
+some subset of the test suite. In order to run a specific test run:
+
+.. code-block:: bash
+
+  pytest unidist/test::test_new_functionality
```
[the remark this answers is not in the document]
Comment (YarShev) [reply docs/developer/contributing.rst:159]: fixed

**Review by prutskov — COMMENTED** [review]
please, change your git config user.name (remove `,`) to be aligned with this contributing guide 😄 
Guide:
image
Your:
image
After, we can merge

**Comment by YarShev** [comment]
Do not see sense, the main point is that there are first name, second name and email address.

**Review by prutskov — APPROVED** [review]

**Pull request merged**